hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTypeConverter.java

```diff
@@ -30,7 +37,52 @@ private HiveTypeConverter() {
 
   }
 
-  public static String convert(Type type) {
+  /**
+   * Converts the Iceberg schema to a Hive schema.
+   * @param schema The original Iceberg schema to convert
+   * @return The Hive column list generated from the Iceberg schema
+   */
+  public static List<FieldSchema> hiveSchema(Schema schema) {
+    return schema.columns().stream()
+        .map(col -> new FieldSchema(col.name(), HiveTypeConverter.convert(col.type()), ""))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Converts the list of Hive FieldSchemas to an Iceberg schema.
+   * <p>
+   * The list should contain the columns and the partition columns as well.
+   * @param fieldSchemas The list of the columns
+   * @return An equivalent Iceberg Schema
+   */
+  public static Schema icebergSchema(List<FieldSchema> fieldSchemas) {
+    List<String> names = new ArrayList<>(fieldSchemas.size());
+    List<TypeInfo> typeInfos = new ArrayList<>(fieldSchemas.size());
+
+    for (FieldSchema col : fieldSchemas) {
+      names.add(col.getName());
+      typeInfos.add(TypeInfoUtils.getTypeInfoFromTypeString(col.getType()));
```

Comment:
Don't we need to copy the comments/doc from `FieldSchema` as well?

Reply:
Thanks @rdsr for the review!

In my previous PRs I have been asked to stick to minimal changes so it is easier to review, so I did not even started to think about adding any change here 😄 (I have been in the reviewing end of the changes and I definitely understand the pain of reviewing the big changes)

That said, your suggestion definitely worth checking. I have some concerns that Hive might disregard the comments set in the schema. I have seen columns with "from deserializer" and I am a little bit concerned about how Hive will handle this.

Would you think this should be another PR, or should we handle this here?

Thanks,
Peter

Reply:
Handling is separately is fine!